test/drivers/helpers/ia32.c

```diff
@@ -5,45 +5,107 @@
  */
 
 #include <unistd.h>
+#include <stdio.h>
+#include <string.h>
 #include <fcntl.h>
 #include <sys/syscall.h>
 #include <sys/types.h>
 #include <sys/socket.h>
 #include <sys/mman.h>
-#include <linux/net.h>        /* Definition of SYS_* constants */
+#include <linux/net.h> /* Definition of SYS_* constants */
 
 #ifdef __NR_openat2
-#include <linux/openat2.h>  /* Definition of RESOLVE_* constants */
+#include <linux/openat2.h> /* Definition of RESOLVE_* constants */
 #endif
 
-int main() {
+#define TRY_1_ARGS_CALL(x, a1)                                                                                         \
```

Comment:
Can't we use a variadic macro?
Ie:
```
#define TRY_CALL(x, ...) \
  if(strncmp(#x, argv[1], sizeof(#x)) == 0)                                                                      \
	{                                                                                                              \
		syscall(x, ##__VA_ARGS__);                                                                                        \
		printf("--> Test_ia32 called '%s'\n", #x);                                                             \
		return 0;                                                                                              \
	}
```

instead of using 3 macros. This should work fine.

Reply:
great idea, thank you! Done